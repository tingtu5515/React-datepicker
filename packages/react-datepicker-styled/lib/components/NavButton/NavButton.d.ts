interface NavButtonProps {
    type: 'next' | 'prev';
}
declare function NavButton({ type }: NavButtonProps): JSX.Element;
export default NavButton;
