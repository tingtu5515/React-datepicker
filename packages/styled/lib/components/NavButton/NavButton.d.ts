interface NavButtonProps {
    type: 'next' | 'prev';
    onClick(): void;
}
declare function NavButton({ type, onClick }: NavButtonProps): JSX.Element;
export default NavButton;
