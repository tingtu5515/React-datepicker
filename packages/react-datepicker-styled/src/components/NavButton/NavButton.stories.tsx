import React from 'react'
import {storiesOf} from '@storybook/react'
import {select} from '@storybook/addon-knobs'
import NavButton from '.'

const iconDirections = {
  left: 'left',
  right: 'right',
}

storiesOf('Components/s', module).add('NavButton', () => (
  <NavButton
    // @ts-ignore
    type={select('Type', iconDirections, 'left')}
  />
))
