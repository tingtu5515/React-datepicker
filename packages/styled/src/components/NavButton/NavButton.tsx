import React from 'react'
import styled from 'styled-components'
import {
  width,
  WidthProps,
  height,
  HeightProps,
  background,
  BackgroundProps,
  space,
  SpaceProps,
  borders,
  BordersProps,
  compose,
} from 'styled-system'
import CaretIcon from '../../icons/CaretIcon'
// eslint-disable-next-line import/no-unresolved
import {NavButtonTheme} from '../../@types/theme'
import useThemeProps from '../../hooks/useThemeProps'
import globalStyles from '../../globalStyles'

interface StyledNavButtonProps
  extends HeightProps,
    WidthProps,
    BackgroundProps,
    SpaceProps,
    BordersProps {}

const composeSyles = compose(
  width,
  height,
  background,
  space,
  borders,
)

const StyledNavButton = styled('button')<StyledNavButtonProps>`
  ${composeSyles}
  display: flex;
  justify-content: center;
  align-items: center;
`

interface NavButtonProps {
  type: 'next' | 'prev'
  onClick(): void
  vertical: boolean
  rtl: boolean
  ariaLabel: string
}

function NavButton({type, onClick, vertical, rtl, ariaLabel}: NavButtonProps) {
  const theme: NavButtonTheme = useThemeProps({
    navButtonWidth: vertical ? '48px' : '30px',
    navButtonHeight: vertical ? '48px' : '30px',
    navButtonBackground: '#ffffff',
    navButtonBorder: '1px solid #929598',
    navButtonPadding: '0',
    navButtonIconHeight: vertical ? '18px' : '11px',
    navButtonIconWidth: vertical ? '28px' : '18px',
    navButtonIconColor: globalStyles.colors.greey,
  })

  function handleMouseUp(e: React.MouseEvent) {
    // @ts-ignore
    e.currentTarget.blur()
  }

  function getDirection() {
    if (type === 'next' && !vertical) {
      return 'right'
    } else if (type === 'next' && vertical) {
      return 'down'
    } else if (type === 'prev' && !vertical) {
      return 'left'
    }

    return 'up'
  }

  return (
    <StyledNavButton
      width={theme.navButtonWidth}
      height={theme.navButtonHeight}
      background={theme.navButtonBackground}
      border={theme.navButtonBorder}
      borderRight={getDirection() === 'up' && !rtl ? 'unset' : theme.navButtonBorder}
      borderLeft={getDirection() === 'up' && rtl ? 'unset' : theme.navButtonBorder}
      p={theme.navButtonPadding}
      type="button"
      aria-label={ariaLabel}
      onClick={onClick}
      onMouseUp={handleMouseUp}
      data-testid="DatepickerNavButton"
    >
      <CaretIcon
        // @ts-ignore
        width={theme.navButtonIconWidth}
        // @ts-ignore
        height={theme.navButtonIconHeight}
        // @ts-ignore
        color={theme.navButtonIconColor}
        direction={getDirection()}
      />
    </StyledNavButton>
  )
}

export default NavButton
