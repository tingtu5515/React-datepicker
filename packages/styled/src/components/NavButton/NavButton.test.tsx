import * as React from 'react'
import {render, fireEvent} from 'react-testing-library'
import NavButton from '.'

test('should match snapshot - prev month', () => {
  const {container} = render(<NavButton vertical={false} type="prev" onClick={jest.fn()} />)
  expect(container).toMatchSnapshot()
})

test('should match snapshot - next month', () => {
  const {container} = render(<NavButton vertical={false} type="next" onClick={jest.fn()} />)
  expect(container).toMatchSnapshot()
})

test('should match snapshot - prev month vertical', () => {
  const {container} = render(<NavButton vertical type="prev" onClick={jest.fn()} />)
  expect(container).toMatchSnapshot()
})

test('should match snapshot - next month vertical', () => {
  const {container} = render(<NavButton vertical type="next" onClick={jest.fn()} />)
  expect(container).toMatchSnapshot()
})

test('should execute onClick callback', () => {
  const onClick = jest.fn()
  const {container} = render(<NavButton vertical={false} type="next" onClick={onClick} />)
  // @ts-ignore
  fireEvent.click(container.firstChild)
  expect(onClick).toHaveBeenCalled()
  // @ts-ignore
  fireEvent.mouseUp(container.firstChild)
  expect(container).toMatchSnapshot()
})
