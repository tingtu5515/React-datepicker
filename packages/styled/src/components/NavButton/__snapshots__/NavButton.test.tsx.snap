// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should execute onClick callback 1`] = `
.c1 {
  -webkit-transform: rotate(-90deg);
  -ms-transform: rotate(-90deg);
  transform: rotate(-90deg);
}

.c0 {
  width: 30px;
  height: 30px;
  background: #ffffff;
  border: 1px solid #929598;
  border-right: 1px solid #929598;
  border-left: 1px solid #929598;
  padding: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div>
  <button
    aria-label="test"
    class="c0"
    data-testid="DatepickerNavButton"
    height="30px"
    type="button"
    width="30px"
  >
    <svg
      class="c1"
      color="#808285"
      height="11px"
      viewBox="0 0 9 6"
      width="18px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M4.058 4.594L1.185 1.72a.312.312 0 1 1 .442-.442L4.5 4.152l2.873-2.873a.312.312 0 1 1 .442.442L4.723 4.812a.316.316 0 0 1-.446 0l-.219-.218z"
        fill="currentColor"
        fill-rule="evenodd"
      />
    </svg>
  </button>
</div>
`;

exports[`should match snapshot - next month 1`] = `
.c1 {
  -webkit-transform: rotate(-90deg);
  -ms-transform: rotate(-90deg);
  transform: rotate(-90deg);
}

.c0 {
  width: 30px;
  height: 30px;
  background: #ffffff;
  border: 1px solid #929598;
  border-right: 1px solid #929598;
  border-left: 1px solid #929598;
  padding: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div>
  <button
    aria-label="test"
    class="c0"
    data-testid="DatepickerNavButton"
    height="30px"
    type="button"
    width="30px"
  >
    <svg
      class="c1"
      color="#808285"
      height="11px"
      viewBox="0 0 9 6"
      width="18px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M4.058 4.594L1.185 1.72a.312.312 0 1 1 .442-.442L4.5 4.152l2.873-2.873a.312.312 0 1 1 .442.442L4.723 4.812a.316.316 0 0 1-.446 0l-.219-.218z"
        fill="currentColor"
        fill-rule="evenodd"
      />
    </svg>
  </button>
</div>
`;

exports[`should match snapshot - next month rtl 1`] = `
.c1 {
  -webkit-transform: rotate(-90deg);
  -ms-transform: rotate(-90deg);
  transform: rotate(-90deg);
}

.c0 {
  width: 30px;
  height: 30px;
  background: #ffffff;
  border: 1px solid #929598;
  border-right: 1px solid #929598;
  border-left: 1px solid #929598;
  padding: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div>
  <button
    aria-label="test"
    class="c0"
    data-testid="DatepickerNavButton"
    height="30px"
    type="button"
    width="30px"
  >
    <svg
      class="c1"
      color="#808285"
      height="11px"
      viewBox="0 0 9 6"
      width="18px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M4.058 4.594L1.185 1.72a.312.312 0 1 1 .442-.442L4.5 4.152l2.873-2.873a.312.312 0 1 1 .442.442L4.723 4.812a.316.316 0 0 1-.446 0l-.219-.218z"
        fill="currentColor"
        fill-rule="evenodd"
      />
    </svg>
  </button>
</div>
`;

exports[`should match snapshot - next month vertical 1`] = `
.c1 {
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
}

.c0 {
  width: 48px;
  height: 48px;
  background: #ffffff;
  border: 1px solid #929598;
  border-right: 1px solid #929598;
  border-left: 1px solid #929598;
  padding: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div>
  <button
    aria-label="test"
    class="c0"
    data-testid="DatepickerNavButton"
    height="48px"
    type="button"
    width="48px"
  >
    <svg
      class="c1"
      color="#808285"
      height="18px"
      viewBox="0 0 9 6"
      width="28px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M4.058 4.594L1.185 1.72a.312.312 0 1 1 .442-.442L4.5 4.152l2.873-2.873a.312.312 0 1 1 .442.442L4.723 4.812a.316.316 0 0 1-.446 0l-.219-.218z"
        fill="currentColor"
        fill-rule="evenodd"
      />
    </svg>
  </button>
</div>
`;

exports[`should match snapshot - next month vertical, rtl 1`] = `
.c1 {
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
}

.c0 {
  width: 48px;
  height: 48px;
  background: #ffffff;
  border: 1px solid #929598;
  border-right: 1px solid #929598;
  border-left: 1px solid #929598;
  padding: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div>
  <button
    aria-label="test"
    class="c0"
    data-testid="DatepickerNavButton"
    height="48px"
    type="button"
    width="48px"
  >
    <svg
      class="c1"
      color="#808285"
      height="18px"
      viewBox="0 0 9 6"
      width="28px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M4.058 4.594L1.185 1.72a.312.312 0 1 1 .442-.442L4.5 4.152l2.873-2.873a.312.312 0 1 1 .442.442L4.723 4.812a.316.316 0 0 1-.446 0l-.219-.218z"
        fill="currentColor"
        fill-rule="evenodd"
      />
    </svg>
  </button>
</div>
`;

exports[`should match snapshot - prev month 1`] = `
.c1 {
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
}

.c0 {
  width: 30px;
  height: 30px;
  background: #ffffff;
  border: 1px solid #929598;
  border-right: 1px solid #929598;
  border-left: 1px solid #929598;
  padding: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div>
  <button
    aria-label="test"
    class="c0"
    data-testid="DatepickerNavButton"
    height="30px"
    type="button"
    width="30px"
  >
    <svg
      class="c1"
      color="#808285"
      height="11px"
      viewBox="0 0 9 6"
      width="18px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M4.058 4.594L1.185 1.72a.312.312 0 1 1 .442-.442L4.5 4.152l2.873-2.873a.312.312 0 1 1 .442.442L4.723 4.812a.316.316 0 0 1-.446 0l-.219-.218z"
        fill="currentColor"
        fill-rule="evenodd"
      />
    </svg>
  </button>
</div>
`;

exports[`should match snapshot - prev month rtl 1`] = `
.c1 {
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
}

.c0 {
  width: 30px;
  height: 30px;
  background: #ffffff;
  border: 1px solid #929598;
  border-right: 1px solid #929598;
  border-left: 1px solid #929598;
  padding: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div>
  <button
    aria-label="test"
    class="c0"
    data-testid="DatepickerNavButton"
    height="30px"
    type="button"
    width="30px"
  >
    <svg
      class="c1"
      color="#808285"
      height="11px"
      viewBox="0 0 9 6"
      width="18px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M4.058 4.594L1.185 1.72a.312.312 0 1 1 .442-.442L4.5 4.152l2.873-2.873a.312.312 0 1 1 .442.442L4.723 4.812a.316.316 0 0 1-.446 0l-.219-.218z"
        fill="currentColor"
        fill-rule="evenodd"
      />
    </svg>
  </button>
</div>
`;

exports[`should match snapshot - prev month vertical 1`] = `
.c1 {
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
}

.c0 {
  width: 48px;
  height: 48px;
  background: #ffffff;
  border: 1px solid #929598;
  border-right: unset;
  border-left: 1px solid #929598;
  padding: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div>
  <button
    aria-label="test"
    class="c0"
    data-testid="DatepickerNavButton"
    height="48px"
    type="button"
    width="48px"
  >
    <svg
      class="c1"
      color="#808285"
      height="18px"
      viewBox="0 0 9 6"
      width="28px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M4.058 4.594L1.185 1.72a.312.312 0 1 1 .442-.442L4.5 4.152l2.873-2.873a.312.312 0 1 1 .442.442L4.723 4.812a.316.316 0 0 1-.446 0l-.219-.218z"
        fill="currentColor"
        fill-rule="evenodd"
      />
    </svg>
  </button>
</div>
`;

exports[`should match snapshot - prev month vertical, rtl 1`] = `
.c1 {
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
}

.c0 {
  width: 48px;
  height: 48px;
  background: #ffffff;
  border: 1px solid #929598;
  border-right: 1px solid #929598;
  border-left: unset;
  padding: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div>
  <button
    aria-label="test"
    class="c0"
    data-testid="DatepickerNavButton"
    height="48px"
    type="button"
    width="48px"
  >
    <svg
      class="c1"
      color="#808285"
      height="18px"
      viewBox="0 0 9 6"
      width="28px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M4.058 4.594L1.185 1.72a.312.312 0 1 1 .442-.442L4.5 4.152l2.873-2.873a.312.312 0 1 1 .442.442L4.723 4.812a.316.316 0 0 1-.446 0l-.219-.218z"
        fill="currentColor"
        fill-rule="evenodd"
      />
    </svg>
  </button>
</div>
`;
